compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -1151,6 +1166,40 @@ pub fn typeid_for_instance<'tcx>(
                 tcx.mk_args_trait(invoke_ty, trait_ref.args.into_iter().skip(1));
             instance.args = instance.args.rebase_onto(tcx, impl_id, abstract_trait_args);
         }
+    } else if tcx.is_closure_like(instance.def_id()) {
+        // We're either a closure or a coroutine. Our goal is to find the trait we're defined on,
+        // instantiate it, and take the type of its only method as our own.
+        let closure_ty = instance.ty(tcx, ty::ParamEnv::reveal_all());
+        if let Some((trait_id, inputs)) = match closure_ty.kind() {
+            ty::Closure(_def_id, args) => {
+                let closure_args = ty::ClosureArgs { args };
```

Comment:
Sorry, was mistaken about `instance.args.type_at(0)`. Just use `instance.args` instead of the closure args.

Also don't do this -- use `args.as_closure()` or `args.as_coroutine()`.